crates/floresta-wire/src/p2p_wire/node.rs

```diff
@@ -364,14 +369,171 @@ where
+                    NodeResponse::Add(true)
+                } else {
+                    info!("Peer {peer_id} ({addr}:{port}) connected once");
+                    NodeResponse::Onetry(true)
+                }
+            }
+            Err(e) => {
+                warn!("{e:?} {addr}:{port}");
+                if insert_peer {
+                    NodeResponse::Add(false)
+                } else {
+                    NodeResponse::Onetry(false)
+                }
+            }
+        };
+
+        responder
+            .send(response)
+            .map_err(|_| WireError::ResponseSendError)?;
+
+        Ok(())
+    }
+
+    /// Handles remove node requests, removing a peer from the node.
+    ///
+    /// This function will try to disconnect from the given address and port, and if it succeeds,
+    /// it will remove the peer from the node and send a bool true to requester. If it fails, it will
+    /// send a bool false to requester.
+    pub async fn handle_disconnect_peer(
+        &mut self,
+        addr: IpAddr,
+        port: u16,
+        responder: oneshot::Sender<NodeResponse>,
+    ) -> Result<(), WireError> {
+        debug!("Trying to remove peer {}:{}", addr, port);
+
+        // First, lets find the requested peer to remove
+        // if it not find any, respond with false
+        let addr_v2 = match addr {
+            IpAddr::V4(a) => AddrV2::Ipv4(a),
+            IpAddr::V6(a) => AddrV2::Ipv6(a),
+        };
+
+        let peer_entry = self.peers.iter().find(|(_, p)| {
+            let ip_match = match p.address {
+                IpAddr::V4(ipv4) => addr_v2 == AddrV2::Ipv4(ipv4),
+                IpAddr::V6(ipv6) => addr_v2 == AddrV2::Ipv6(ipv6),
+            };
+            ip_match && p.port == port
+        });
+
+        let (peer_id, _) = match peer_entry {
+            Some(entry) => (*entry.0, entry.1),
+            None => {
+                warn!("Peer {addr}:{port} not found in the list of known peers");
+                responder
+                    .send(NodeResponse::Remove(false))
+                    .map_err(|_| WireError::ResponseSendError)?;
+                return Ok(());
+            }
+        };
+
+        // If the requested peer is found, then we need send a NodeResponse::Shutdown.
+        // After shutting down, we need to wait for the peer to disconnect and then
+        // remove it from the node. If some disconnect failure happens, we need to
+        // respond with false. If it's successful, we need to remove it from the peers list.
+        // If some kind of error happens during the remotion, we need to respond with false.
+        // If the remotion happens well, we need to drop the peer channel and respond with true.
+        let response = match self.send_to_peer(peer_id, NodeRequest::Shutdown).await {
+            Ok(_) => {
+                if let Some((_, address, _)) = self.added_peers.remove(&peer_id) {
```

Comment:
Is all this error handling required here? If you use the `?` operator (e.g. `self.do_something()?;` you propagate the error to the caller, and you only need to handle it once in the caller.